Include/internal/pycore_bitutils.h

```diff
@@ -82,6 +84,45 @@ _Py_bswap64(uint64_t word)
 }
 
 
+// Population count: count the number of 1's in 'u'
+// (number of bits set to 1), also known as the hamming weight.
+//
+// Implementation note. CPUID is not used, to test if x86 POPCNT instruction
+// can be used, to keep the implementation simple. For example, Visual Studio
+// __popcnt() is not used this reason. The clang and GCC builtin function can
+// use the x86 POPCNT instruction if the target architecture has SSE4a or
+// newer.
+static inline int
+_Py_popcount32(uint32_t x)
+{
+#if (defined(__clang__) || defined(__GNUC__)) && SIZEOF_INT == 4
+    return __builtin_popcount(x);
+#elif (defined(__clang__) || defined(__GNUC__)) && SIZEOF_INT == 8
```

Comment:
Should this be `SIZEOF_LONG == 4` instead of `SIZEOF_INT == 8`? We want to count bits in a `uint32_t`, and `__builtin_popcountl` counts bits in an unsigned long, so we only want to use it if `uint32_t` matches `unsigned long`.

Reply:
Actually, I'd suggest dropping the `SIZEOF_INT == 8` check altogether. The C standard guarantees that `unsigned long` will always be big enough to hold a `uint32_t` value without losing information, so using `__builtin_popcountl` should always be safe.

The check in the first branch could be weakened to `SIZEOF_INT >= 4`, but that's unlikely to make a difference in practice.

Reply:
I rewrote the code to always use __builtin_popcountl if SIZEOF_INT < 4. I added a comment explaining why it's safe.